src/main/java/org/sciborgs1155/robot/Robot.java

```diff
@@ -56,13 +58,17 @@ public class Robot extends CommandRobot implements Logged {
   private final CommandXboxController operator = new CommandXboxController(OI.OPERATOR);
   private final CommandXboxController driver = new CommandXboxController(OI.DRIVER);
 
+  @Log.NT private boolean noteDetected = false;
+  private EventLoop intakeTriggerPoller = new EventLoop();
+  private Trigger intakeTrigger = new Trigger(intakeTriggerPoller, () -> noteDetected);
+
   // SUBSYSTEMS
   private final Drive drive = Drive.create();
 
   private final Intake intake =
       switch (Constants.ROBOT_TYPE) {
         case CHASSIS -> Intake.none();
-        default -> Intake.create();
+        default -> Intake.create((b) -> noteDetected = b);
```

Comment:
also for the record I see this as worse than creating a method in intake that addPeriodic calls